src/DelegationManagement.sol

```diff
@@ -1,183 +1,317 @@
 // SPDX-License-Identifier: MIT
 
-/** 
+/**
  *
- *  @title: Delegation Management Contract   
+ *  @title: Delegation Management Contract
  *  @date: 21-Dec-2022 @ 10:30
- *  @version: 4.23 
+ *  @version: 4.23
  *  @notes: This is a experimental contract for delegation registry
  *  @author: skynet2030 (skyn3t2030)
  *
  */
 
-pragma solidity >=0.6.0 <=0.8.0;
+pragma solidity 0.8.17;
 
 contract delegationManagement {
-
     // Variable declarations
-    uint256 useCaseCounter; 
+    uint256 useCaseCounter;
 
     // Mapping declarations
-    mapping (bytes32 => bool) public registeredDelegation;
-    mapping (bytes32 => uint256) public delegationToCounterPerHash;
-    mapping (bytes32 => uint256) public delegationFromCounterPerHash;
+    mapping(bytes32 => bool) public registeredDelegation;
+    mapping(bytes32 => uint256) public delegationToCounterPerHash;
+    mapping(bytes32 => uint256) public delegationFromCounterPerHash;
 
     // Struct declaration
     struct delegationAddresses {
         address mainAddress;
+        uint96 registeredDate;
         bytes32 delegationGlobalHash;
         bytes32 delegationToHash;
         bytes32 delegationFromHash;
         address collectionAddress;
+        uint96 expiryDate;
         address delegationAddress;
-        uint256 registeredDate;
-        uint256 expiryDate;
-        uint256 useCase;
+        uint96 useCase;
     }
 
     // bytes32 mappings with arrays
-    mapping (bytes32 => delegationAddresses[]) public delegateToHashes;
-    mapping (bytes32 => delegationAddresses[]) public delegateFromHashes;
+    mapping(bytes32 => delegationAddresses[]) public delegateToHashes;
+    mapping(bytes32 => delegationAddresses[]) public delegateFromHashes;
 
     // Events declaration
 
-    event registerDelegation(address indexed from, address indexed collectionAddress, address indexed delegationAddress, uint256 useCase);
-    event revokeDelegation(address indexed from, address indexed collectionAddress, address indexed delegationAddress, uint256 useCase);
-    event updateDelegation(address indexed from, address indexed collectionAddress, address olddelegationAddress, address indexed newdelegationAddress, uint256 useCase);
-    
+    event registerDelegation(
```

Comment:
Beautifying the event declarations should also be a different commit

Reply:
I agree